networks/monero/src/merkle.rs

```diff
@@ -2,14 +2,13 @@ use std_shims::vec::Vec;
 
 use crate::primitives::keccak256;
 
-pub(crate) fn merkle_root(root: [u8; 32], leafs: &[[u8; 32]]) -> [u8; 32] {
+/// Calculates the merkel root of the given tree. Equivalent to `tree_hash` in monero-core.
+pub fn merkle_root(leafs: &[[u8; 32]]) -> [u8; 32] {
   match leafs.len() {
-    0 => root,
```

Comment:
We used to handle no leaves being passed. Now no hashes falls through to the n-leaves case. Is that correct?

I doubt it as our n-leaf code pads to the nearest power of 2 which is assumed to be 4 as we assumed exceptional handling for 0/1/2.

Reply:
Ah yeah it will panic, I guess the best thing to do would be to explicitly panic, `tree_hash` doesn't allow a 0 length input: https://github.com/monero-project/monero/blob/893916ad091a92e765ce3241b94e706ad012b62a/src/crypto/tree-hash.c#L62

Reply:
That would've been so easy for me to check if there was a link to `tree_hash` present :p

*I didn't add one when I wrote this, I don't legitimately want to hold this against you.

I'd vote we keep the existing explicit argument (potentially renamed from root to first_leaf?) making this unreachable, or return an Option. We can document the panic but I'd consider that a downgrade. Even the Option would be but it isn't such a downgrade I wouldn't agree to it if you believe avoiding these allocations so worthwhile.